src/syngen/streamlit_app.py

```diff
@@ -0,0 +1,318 @@
+                             f"{slugify(datetime.now().strftime('%Y-%m-%d %H:%M:%S'))}.log")
+        self.path_to_report = f"model_artifacts/tmp_store/{self.sl_table_name}/draws/accuracy_report.html"
+
+    def set_parameters(self, epochs, size_limit, print_report):
+        """
+        Set parameters for a model training and data generation
+        """
+        self.epochs = epochs
+        self.size_limit = size_limit
+        self.print_report = print_report
+
+    def set_env_variables(self):
+        """
+        Set environment variables
+        """
+        os.environ["PATH_TO_GENERATED_DATA"] = self.path_to_generated_data
+        os.environ["PATH_TO_REPORT"] = self.path_to_report
+        os.environ["SUCCESS_LOG_FILE"] = self.path_to_logs
+        os.environ["PRINT_REPORT"] = "True" if self.print_report else str()
+        os.environ["TABLE_NAME"] = self.sl_table_name
+
+    def set_logger(self):
+        """
+        Set a logger in order to see logs in the Streamlit app,
+        and collect log messages with the log level - 'INFO' in a log file
+        """
+        logger.add(self.file_sink, level="INFO")
+        logger.add(self.log_sink)
+
+    def log_sink(self, message):
+        """
+        Put log messages to a log queue
+        """
+        log_message = fetch_log_message(message)
+        self.log_queue.put(log_message)
+
+    def file_sink(self, message):
+        """
+        Write log messages to a log file
+        """
+        os.makedirs(os.path.dirname(self.path_to_logs), exist_ok=True)
+        with open(self.path_to_logs, "a") as log_file:
+            log_message = fetch_log_message(message)
+            log_file.write(log_message + "\n")
+
+    def train_model(self):
+        """
+        Launch a model training
+        """
+        try:
+            self.set_logger()
+            logger.info("Starting model training...")
+            settings = {
+                "source": self.file_path,
+                "epochs": self.epochs,
+                "row_limit": 10000,
+                "drop_null": False,
+                "batch_size": 32,
+                "print_report": False
+            }
+            worker = Worker(
+                table_name=self.table_name,
+                settings=settings,
+                metadata_path=None,
+                log_level="INFO",
+                type_of_process="train"
+            )
+            worker.launch_train()
+            logger.info("Model training completed")
+        except Exception as e:
+            logger.error(f"Error during train: {e}")
+            self.log_error_queue.put(f"Error during train: {e}")
+
+    def infer_model(self):
+        """
+        Launch a data generation
+        """
+        try:
+            logger.info("Starting data generation...")
+            settings = {
+                "size": self.size_limit,
+                "batch_size": 32,
+                "run_parallel": False,
+                "random_seed": None,
```

Comment:
During infer process we have fixed random_seed = 1 by default.
Do we need to set it here for reproducibility?